### PR TITLE
BREAKING CHANGE: use Navigator interface instead of Document

### LINKED ISSUE #24: `document.monetization` vs global
From web-monetization created by [marcoscaceres](https://github.com/marcoscaceres): adrianhopebailie/web-monetization#23  

Wondering what the rationale was for attaching monetization to the `Document` object? It doesn't really feel like the right place to attach this, as monetization doesn't itself affect the document or have anything specifically to do with the document interface. We should consider maybe having monetization either on Navigator or its own `Monetization` namespace.

### PR DESCRIPTION
**Pull request by marcoscaceres**
Closes #24

### CODE REVIEW

**Review by AlexLakatos — APPROVED** [review]
This seems to be one of the most contended issues. Syntactically, I agree with it, Monetization should live on the Navigator instead of the Document. Window is just a "couldn't come up with anything else" place, and prone to clashing with user code.

I wonder how this will clash with the existing Web Monetization provider extension implementation. @sharafian any strong opinions against this?

**Comment by sharafian** [comment]
using navigator is alright with me, assuming that there aren't any quirks that come up in the implementation. It's also nice because we can keep the document interface around until enough users have migrated off of it, and implement these new breaking changes on the navigator interface

**Comment by AlexLakatos** [comment]
Perfect! That also makes #209 easier to approve then!

File: static/specification.html [hunk 1]
```diff
@@ -236,7 +237,7 @@ <h2>
             with a non-zero amount, the provider notifies the browser:
               <ul>
                 <li>The <a>user agent</a> sets
-                <code>document.monetization.state</code> to
+                <code>navigator.monetization.state</code> to
```
Comment (sublimator) [inline static/specification.html:240]: What states do we imagine this having and how will they be used in if we are collapsing the events, and in particular removing the stop event? 

Currently we have `pending | started | stopped`
We'll set the state to `stopped` but not have a corresponding event ?
We'll no longer have a `pending` event but set the `state` ?

Comment (sublimator) [reply static/specification.html:240]: This reminds me of a conversation we had internally on slack with Matt Baer from write.as 
He wasn't happy just using a timeout to determine a user wasn't willing or able to monetize

Comment (marcoscaceres) [reply static/specification.html:240]: Hey, so we can add as many events as we need. The collapsing of events is just to merge the interface, but it doesn't need to limit the number of states we can have. 

>  We'll set the state to stopped but not have a corresponding event ?

No, let me clarify with an example 

```
navigator.monetization.state; // changes to "stopped"
navigator.monetization.onstatechage = event => {
     switch(navigator.monetization.state):
            case "stopped":
                 // do stuff..
            case "started":
                 // do stuff..
            case "whatever-we-need":
                // do stuff..
            default: 
                // do default stuff..
}
```

Comment (sublimator) [reply static/specification.html:240]: Got it! 
In the latest extension, I mapped pending/started to the proposed 'interactive', and stopped to 'idle'
Would be good if we can agree on some terms @ https://github.com/WICG/webmonetization/issues/220

I'm pretty sure @sharafian would like to see as little change as possible

**Comment by marcoscaceres** [comment]
Duplicate of #193

**Pull request closed without merging**